packages/dev/core/src/Materials/meshDebugPluginMaterial.ts

```diff
@@ -0,0 +1,473 @@
+     * @param _scene Scene
+     * @param _mesh Mesh associated with material
+     */
+    public prepareDefines(defines: MeshDebugDefines, scene: Scene, mesh: AbstractMesh) {
+        if ((this._mode == MeshDebugMode.VERTICES || this._mode == MeshDebugMode.TRIANGLES || this._mode == MeshDebugMode.TRIANGLES_VERTICES) && !mesh.isVerticesDataPresent("dbg_initialPass")) {
+            Logger.Warn("For best results with TRIANGLES, TRIANGLES_VERTICES, or VERTICES modes, please use MeshDebugPluginMaterial.prepareMeshForTrianglesAndVerticesMode() on mesh.", 1);
+        }
+        defines.dbg_MODE = this._mode!;
+        defines.dbg_MULTIPLYDEBUG = this._options.multiplyDebug!;
+    }
+
+    /**
+     * Get the shader attributes
+     * @param attributes Array of attributes
+     */
+    public getAttributes(attr: string[]) {
+        attr.push("dbg_initialPass");
+    }
+
+    /**
+     * Get the shader uniforms
+     * @returns Uniforms
+     */
+    public getUniforms() {
+        return {
+            ubo: [
+                { name: "dbg_shadedDiffuseColor", size: 3, type: "vec3" },
+                { name: "dbg_shadedSpecularColorPower", size: 4, type: "vec4" }, // shadedSpecularColor, shadedSpecularPower
+                { name: "dbg_thicknessRadiusUvDistance", size: 4, type: "vec4" }, // wireframeThickness, vertexRadius, uvScale, and distanceScale
+                { name: "dbg_wireframeColor1", size: 3, type: "vec3" },
+                { name: "dbg_wireframeColor2", size: 3, type: "vec3" },
+                { name: "dbg_uvColor1", size: 3, type: "vec3" },
+                { name: "dbg_uvColor2", size: 3, type: "vec3" },
+                { name: "dbg_materialColor", size: 3, type: "vec3" },
+            ],
+            fragment: `
+                uniform vec3 dbg_shadedDiffuseColor;
+                uniform vec4 dbg_shadedSpecularColorPower;
+                uniform vec4 dbg_thicknessRadiusUvDistance;
+
+                #if dbg_MODE == 0 || dbg_MODE == 1
+                    uniform vec3 dbg_wireframeColor1;
+                #endif
+                #if dbg_MODE == 1
+                    uniform vec3 dbg_wireframeColor2;
+                #elif dbg_MODE == 2 || dbg_MODE == 3
+                    uniform vec3 dbg_uvColor1;
+                    uniform vec3 dbg_uvColor2;
+                #elif dbg_MODE == 5
+                    uniform vec3 dbg_materialColor;
+                #endif
+            `,
+        };
+    }
+
+    /**
+     * Bind the uniform buffer
+     * @param uniformBuffer Uniform buffer
+     */
+    public bindForSubMesh(uniformBuffer: UniformBuffer): void {
+        if (this._isEnabled) {
+            uniformBuffer.updateVector3("dbg_shadedDiffuseColor", this._options.shadedDiffuseColor!);
+            uniformBuffer.updateVector4("dbg_shadedSpecularColorPower", new Vector4(this._options.shadedSpecularColor!.x, this._options.shadedSpecularColor!.y, this._options.shadedSpecularColor!.z, this._options.shadedSpecularPower!));

```

Comment:
You should use `updateFloat4` instead of `updateVector4`, to avoid creating a new `Vector4` each time.

Same thing for the line below.